File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeRemovedPaymentForCraftMessage.scala





















// Generated on 05/03/2013 10:34:55
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeRemovedPaymentForCraftMessage extends MessageDeserializer {
  val messageId = 6031

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeRemovedPaymentForCraftMessage.messageId

    val onlySuccess: Boolean
    val objectUID: Int

    def serialize(buf: Buffer) {
      buf.writeBoolean(onlySuccess)
      buf.writeInt(objectUID)
    }

  }

  protected def create(onlySuccess: Boolean, objectUID: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val onlySuccess = buf.readBoolean()
    val objectUID = buf.readInt()

    create(onlySuccess, objectUID)
  }

}

object ExchangeRemovedPaymentForCraftMessage extends ExchangeRemovedPaymentForCraftMessage {
  case class MessageImpl(onlySuccess: Boolean, objectUID: Int) extends Message
  type Target = MessageImpl

  override def create(onlySuccess: Boolean, objectUID: Int) = MessageImpl(onlySuccess, objectUID)

  def apply(onlySuccess: Boolean, objectUID: Int) = create(onlySuccess, objectUID)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeRemovedPaymentForCraftMessage.java


// Generated on 05/08/2013 19:37:54
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeRemovedPaymentForCraftMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6031;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean onlySuccess;
    public int objectUID;
    
    public ExchangeRemovedPaymentForCraftMessage() { }
    
    public ExchangeRemovedPaymentForCraftMessage(boolean onlySuccess, int objectUID) {
        this.onlySuccess = onlySuccess;
        this.objectUID = objectUID;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeBoolean(onlySuccess);
        buf.writeInt(objectUID);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        onlySuccess = buf.readBoolean();
        objectUID = buf.readInt();
        if (objectUID < 0)
            throw new RuntimeException("Forbidden value on objectUID = " + objectUID + ", it doesn't respect the following condition : objectUID < 0");
    }
    
}
